src/NHibernate/Linq/Functions/CompareGenerator.cs

```diff
@@ -44,9 +44,17 @@ internal static bool IsCompareMethod(MethodInfo methodInfo)
 				return true;
 
 			// This is .Net 4 only, and in the System.Data.Services assembly, which we don't depend directly on.
-			return methodInfo != null && methodInfo.Name == "Compare" &&
+			if (methodInfo != null && methodInfo.Name == "Compare" &&
 				   methodInfo.DeclaringType != null &&
-				   methodInfo.DeclaringType.FullName == "System.Data.Services.Providers.DataServiceProviderMethods";
+				   methodInfo.DeclaringType.FullName == "System.Data.Services.Providers.DataServiceProviderMethods")
+				return true;
+
+			if (methodInfo != null && methodInfo.Name == "CompareString" &&
```

Comment:
Ok. Is this work-around fine for now, or should NHibernate depend on the alpha package?  Also, include `https:\\` in your links...